contrib/geoToolbox.lua

```diff
@@ -0,0 +1,590 @@
+--[[
+    This file is part of darktable,
+    copyright (c) 2016 Tobias Jakobs
+
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    darktable is distributed in the hope that it will be useful,
+     but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+]]
+--[[
+
+USAGE
+* require this script from your main lua file
+  To do this add this line to the file .config/darktable/luarc: 
+require "geoToolbox"
+
+* it creates a new geoToolbox lighttable module
+]]
+
+local dt = require "darktable"
+local df = require "lib/dtutils.file"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0},{4,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("geoToolbox",dt.configuration.config_dir.."/lua/")
+
+local function _(msgid)
+    return gettext.dgettext("geoToolbox", msgid)
+end
+
+-- <GUI>
+local labelDistance = dt.new_widget("label")
+labelDistance.label = "Distance:"
+
+local labelCopyGPSlat = dt.new_widget("check_button")
+{
+  label = "latitude:", 
+  value = true
+}
+local labelCopyGPSlon = dt.new_widget("check_button")
+{
+  label = "longitude:", 
+  value = true
+}
+local labelCopyGPSele = dt.new_widget("check_button")
+{
+  label = "elevation:", 
+  value = true
+}
+-- </GUI> 
+
+local function selectWithGPS()
+   local selection = {}
+   for _,image in ipairs(dt.database) do
+      if (image.longitude and image.latitude) then
+         table.insert(selection,image)
+      end
+   end
+   dt.gui.selection(selection)
+end
+
+local function selectWithoutGPS()
+   local selection = {}
+   for _,image in ipairs(dt.database) do
+      if (not image.longitude and not image.latitude) then
+         table.insert(selection,image)
+      end
+   end
+   dt.gui.selection(selection)
+end
+
+-- This is used for older images in the DB
+function isnan(x) return x ~= x end
+
+-- Function from:
+-- https://forums.coronalabs.com/topic/29019-convert-string-to-date/
+-- This looks much scarier than it really is. The goal is to turn that string 
+-- into a Unix timestamp (number of seconds since Jan 1, 1970, the standard 
+-- used by most systems). So we use the string.match() method to fetch the 
+-- various date and time parts into their own variables: xyear, xmonth,etc.
+function makeTimeStamp(dateString)
```

Comment:
What do you think about moving this into a new lib/dtutils.datetime helper library?